src/components/ResourceBrowser/ResourceList/BaseResourceList.tsx

```diff
@@ -107,8 +121,18 @@ const BaseResourceListContent = ({
         getSelectedIdentifiersCount,
     } = useBulkSelection<Record<number, K8sResourceDetailDataType>>()
 
+    const headers = useMemo(() => {
+        if (!isNodeListing) {
+            return resourceList?.headers ?? []
+        }
+
+        const visibleColumnsSet = new Set(visibleColumns)
+
+        return resourceList?.headers.filter((header) => visibleColumnsSet.has(header)) ?? []
```

Comment:
Missing null check, please create a variable for resourceList?.headers ?? [] and re-use it above early return as well